docs(module-doc): update content about input config (#4587)
web-infra-dev/modern.js: packages/document/module-doc/docs/en/api/config/build-config.mdx

**Array usage:**

> In `bundle` mode, the following configurations will be built using `src/index.ts` and `src/index2.ts` as entry points. The usage of catalogs is not supported.

Reviewer: ```suggestion
In `bundle` mode, the following configurations will be built using `src/index.ts` and `src/index2.ts` as entry points. The `bundle` mode does not support configuring `input` as a directory.
```
